routing/probability_estimator.go

```diff
@@ -155,22 +156,23 @@ func (p *probabilityEstimator) calculateProbability(
 		return nodeProbability
 	}
 
-	// For successes, we have a fixed (high) probability. Those pairs
-	// will be assumed good until proven otherwise.
-	if lastPairResult.Success {
-		return p.prevSuccessProbability
+	// For successes, we have a fixed (high) probability. Those pairs will
+	// be assumed good until proven otherwise. Amt is never zero, so this
+	// clause is never executed when lastPairResult.SuccessAmt is zero.
+	if amt <= lastPairResult.SuccessAmt {
+		return prevSuccessProbability
```

Comment:
if this is a global constant now, maybe delete the field on the `probabilityEstimator`